projects/client/RabbitMQ.Client/src/client/impl/AutorecoveringConnection.cs

```diff
@@ -799,18 +852,24 @@ private void RecoverModels()
 
         private void RecoverQueues()
         {
-            lock (RecordedQueues)
+            IDictionary<string, RecordedQueue> recordedQueuesCopy = null;
+            lock (_recordedEntitiesLock)
+            {
+                recordedQueuesCopy = _recordedQueues.ToDictionary(entry => entry.Key, entry => entry.Value);
+            }
+
+            foreach (KeyValuePair<string, RecordedQueue> pair in recordedQueuesCopy)
             {
-                foreach (KeyValuePair<string, RecordedQueue> pair in RecordedQueues)
+                string oldName = pair.Key;
+                RecordedQueue rq = pair.Value;
+
+                try
                 {
-                    string oldName = pair.Key;
-                    RecordedQueue rq = pair.Value;
+                    rq.Recover();
+                    string newName = rq.Name;
 
-                    try
+                    if (!oldName.Equals(newName))
```

Comment:
This matches the Java client behavior. It will change the behavior in this library by preventing events from being fired that would have been fired previously when the name stays the same. @bording @michaelklishin @acogoluegnes @kjnilsson I am assuming that we wish to stay as close to the Java client's behavior as possible.

Reply:
I'm assuming that if I back-port this to `5.2.0` I will have to revert to the previous behavior.